test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -206,7 +207,7 @@ def test_zero_grad(self):
 class TestZeroRedundancyOptimizerDistributed(TestZeroRedundancyOptimizer):
     @property
     def world_size(self):
-        return max(2, torch.cuda.device_count())
+        return min(4, max(2, torch.cuda.device_count()))
```

Comment:
run the tests if more than 4 gpus available, but bound the world size, I figured that was better than skipping it